server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/RemoteShardsBalancer.java

```diff
@@ -432,6 +438,10 @@ private void tryAllocateUnassignedShard(Queue<RoutingNode> nodeQueue, ShardRouti
                     }
                     nodeQueue.offer(node);
                 } else {
+                    if (nodeLevelDecision.type() == Decision.Type.THROTTLE) {
```

Comment:
wouldn't the nodeLevelDecision always be Decision.Type.THROTTLE if allocationDesction.type() is THROTTLE . Since if a node is throttling once shard it would also throttle other shards

then we dont need to call `canAllocateAnyShardToNode` if allocateDecision.type() == Decision.Type.THROTTLE

Reply:
I think that shard level decision and node level decision can be different